src/main/kotlin/com/virtuslab/dokka/site/processors.kt

```diff
@@ -200,19 +200,23 @@ class SitePagesCreator(cxt: DokkaContext) : BaseStaticSiteProcessor(cxt) {
                 val context = RenderingContext(emptyMap(), layouts)
                 if (from.isDirectory) {
                     children.find { it.isIndexPage }?.resolved ?: EmptyResolvedPage
-                } else templateFile.resolve(context)
+                } else if (templateFile.layout() != null ){
+                    templateFile.layout().let { layouts[it] }
+                        ?. resolve(context.nest(templateFile.rawCode, from.absolutePath, emptyList()))
```

Comment:
Do we fill variables? We need to do that otherwise we loose `content` from original page when we have double nesting.

Reply:
simply added to rendering context then